packages/datadog-plugin-aws-sdk/test/dynamodb.spec.js

```diff
@@ -0,0 +1,878 @@
+              try {
+                if (typeof dynamo.deleteTable({}).promise === 'function') {
+                  await dynamo.deleteTable({ TableName: tableName }).promise()
+                  await dynamo.waitFor('tableNotExists', { TableName: tableName }).promise()
+                } else {
+                  await new Promise((resolve, reject) => {
+                    dynamo.deleteTable({ TableName: tableName }, (err) => {
+                      if (err && err.code !== 'ResourceNotFoundException') {
+                        reject(err)
+                      } else {
+                        resolve()
+                      }
+                    })
+                  })
+                }
+              } catch (err) {
+                if (err.code !== 'ResourceNotFoundException') {
+                  throw err
+                }
+              }
+            }
+          }
+
+          const createTable = async (params) => {
+            if (dynamoClientName === '@aws-sdk/client-dynamodb') {
+              await dynamo.createTable(params)
+            } else {
+              if (typeof dynamo.createTable({}).promise === 'function') {
+                await dynamo.createTable(params).promise()
+              } else {
+                await new Promise((resolve, reject) => {
+                  dynamo.createTable(params, (err, data) => {
+                    if (err) reject(err)
+                    else resolve(data)
+                  })
+                })
+              }
+            }
+          }
+
+          // Delete existing tables
+          await deleteTable(oneKeyTableName)
+          await deleteTable(twoKeyTableName)
+
+          // Create tables
+          await createTable({
+            TableName: oneKeyTableName,
+            KeySchema: [{ AttributeName: 'name', KeyType: 'HASH' }],
+            AttributeDefinitions: [{ AttributeName: 'name', AttributeType: 'S' }],
+            ProvisionedThroughput: { ReadCapacityUnits: 5, WriteCapacityUnits: 5 }
+          })
+
+          await createTable({
+            TableName: twoKeyTableName,
+            KeySchema: [
+              { AttributeName: 'id', KeyType: 'HASH' },
+              { AttributeName: 'binary', KeyType: 'RANGE' }
+            ],
+            AttributeDefinitions: [
+              { AttributeName: 'id', AttributeType: 'N' },
+              { AttributeName: 'binary', AttributeType: 'B' }
+            ],
+            ProvisionedThroughput: { ReadCapacityUnits: 5, WriteCapacityUnits: 5 }
+          })
+        })
+
+        after(async () => {
+          await resetLocalStackDynamo()
+          return agent.close({ ritmReset: false })
+        })
+
+        describe('span pointers', () => {
+          beforeEach(() => {
+            DynamoDb.dynamoPrimaryKeyConfig = null
+            delete process.env.DD_AWS_SDK_DYNAMODB_TABLE_PRIMARY_KEYS
+          })
+
+          function testSpanPointers ({ expectedLength = 1, expectedHashes, operation }) {
```

Comment:
nit: shouldn't expected length be equal to the length of expected hashes if that is an array or 1 otherwise? does it need to be a parameter?

Reply:
or, maybe even better, we can either take `expectedHash` or `expectedHashes` as parameters, failing if we get both, and then do the right thing in this code? the tests below would read a little more clearly then.

Reply:
Yes good idea! I think it would be simplest to just remove the `expectedLength` parameter, and then we can just. do this:
```
let expectedLength = 0
if (expectedHashes) {
  expectedLength = Array.isArray(expectedHashes) ? expectedHashes.length : 1
}
```